### PR TITLE
Feature/Toast error on invalid axes selection and default value options

### PR DESCRIPTION
**Pull request by rinku-kumar-psl**
### Description
[Describe what this change achieves]
OpenSearch-project#666

- Default axes selected on Data config value options
- Error toast If the user selected any invalid or empty axis
- Error toast if the user trying to save the visualization, If f user selected any invalid or empty axis

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: dashboards-observability/public/components/custom_panels/__tests__/__snapshots__/custom_panel_table.test.tsx.snap [hunk 1]
```diff
@@ -1894,7 +1894,7 @@ exports[`Panels Table Component renders panel table container 1`] = `
                                           <div
                                             className="euiTableCellContent euiTableCellContent--overflowingContent"
                                           >
-                                            11/04/2021 06:10 PM
+                                            11/04/2021 11:40 PM
```
Comment (abasatwar) [inline dashboards-observability/public/components/custom_panels/__tests__/__snapshots__/custom_panel_table.test.tsx.snap:1897]: revert this change.

File: dashboards-observability/public/components/custom_panels/__tests__/__snapshots__/custom_panel_table.test.tsx.snap [hunk 2]
```diff
@@ -1926,7 +1926,7 @@ exports[`Panels Table Component renders panel table container 1`] = `
                                           <div
                                             className="euiTableCellContent euiTableCellContent--overflowingContent"
                                           >
-                                            11/02/2021 07:26 PM
+                                            11/03/2021 12:56 AM
```
Comment (abasatwar) [inline dashboards-observability/public/components/custom_panels/__tests__/__snapshots__/custom_panel_table.test.tsx.snap:1929]: same here and follow the approach we ae following for other PRs

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx [hunk 3]
```diff
@@ -226,4 +245,4 @@ export const ConfigPanel = ({ visualizations, setCurVisId }: any) => {
       </EuiFlexGroup>
     </>
   );
-};
+};
```
Comment (abasatwar) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx:248]: add empty line

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/index.tsx [hunk 4]
```diff
@@ -39,6 +40,8 @@ export const ExplorerVisualizations = ({
   handleRemoveField,
   visualizations,
   handleOverrideTimestamp,
+  updateIsDataConfigOptionsInvalid
+
```
Comment (abasatwar) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/index.tsx:44]: remove empty space

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx [hunk 5]
```diff
@@ -98,7 +117,7 @@ export const ConfigPanel = ({ visualizations, setCurVisId }: any) => {
     } catch (e) {
       setToast(`Invalid visualization configurations. error: ${e.message}`, 'danger');
     }
-  }, [tabId, vizConfigs, changeVisualizationConfig, dispatch, setToast, curVisId]);
+  }, [tabId, vizConfigs, changeVisualizationConfig, dispatch, setToast, curVisId, vizConfigs]);
```
Comment (abasatwar) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx:120]: remove double entry of vizConfigs

File: dashboards-observability/public/components/notebooks/components/paragraph_components/__tests__/__snapshots__/para_input.test.tsx.snap [hunk 6]
```diff
@@ -274,7 +274,7 @@ exports[`<para_input /> spec renders the visualization component 1`] = `
                           data-test-subj="superDatePickerstartDatePopoverButton"
                           title="2020-07-21T18:37:44.710Z"
                         >
-                          2020-07-21T18:37:44+00:00
+                          2020-07-22T00:07:44+05:30
```
Comment (spattnaik) [inline dashboards-observability/public/components/notebooks/components/paragraph_components/__tests__/__snapshots__/para_input.test.tsx.snap:277]: As mentioned by Amruta earlier, try not updating dates in snapshot.

**Review by spattnaik — CHANGES_REQUESTED** [review]
Looks good from coding perspective. Check for comments from Amruta and check the date fields in snapshots.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/index.tsx (hunk 4 above)
Comment (rinku-kumar-psl) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/index.tsx:44]: Done.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx (hunk 5 above)
Comment (rinku-kumar-psl) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx:120]: Done

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx (hunk 3 above)
Comment (rinku-kumar-psl) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx:248]: Done

File: dashboards-observability/public/components/custom_panels/__tests__/__snapshots__/custom_panel_table.test.tsx.snap (hunk 1 above)
Comment (rinku-kumar-psl) [reply dashboards-observability/public/components/custom_panels/__tests__/__snapshots__/custom_panel_table.test.tsx.snap]: Done

File: dashboards-observability/public/components/custom_panels/__tests__/__snapshots__/custom_panel_table.test.tsx.snap (hunk 2 above)
Comment (rinku-kumar-psl) [reply dashboards-observability/public/components/custom_panels/__tests__/__snapshots__/custom_panel_table.test.tsx.snap]: Done

File: dashboards-observability/public/components/notebooks/components/paragraph_components/__tests__/__snapshots__/para_input.test.tsx.snap (hunk 6 above)
Comment (rinku-kumar-psl) [reply dashboards-observability/public/components/notebooks/components/paragraph_components/__tests__/__snapshots__/para_input.test.tsx.snap]: Done. Thanks Subrat!

**Review by spattnaik — APPROVED** [review]
Check the comment and let me know your inputs. Other things look good.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx [hunk 7]
```diff
@@ -89,16 +96,27 @@ export const ConfigPanel = ({ visualizations, setCurVisId, updateIsDataConfigOpt
...
   );
 
-  // To check , If user empty any of the value options
-  const isInvalidValueOptionConfig = useMemo(() => !(vizConfigs.dataConfig && (vizConfigs.dataConfig?.valueOptions?.xaxis?.length !== 0
-    && vizConfigs.dataConfig?.valueOptions?.yaxis?.length !== 0)), [vizConfigs])
+  // To check, If user empty any of the value options
+  const isValidValueOptionConfigSelected = useMemo(() => {
+    const valueOptions = vizConfigs.dataConfig?.valueOptions;
+    const { Bar, Line, TimeSeries, Histogram, Pie, TreeMap, Gauge, HeatMap } = visChartTypes;
+    return valueOptions && (
```
Comment (abasatwar) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx:103]: Can we create variables to store multiple conditions to make this code more readable?

File: dashboards-observability/common/constants/shared.ts [hunk 8]
```diff
@@ -73,7 +97,8 @@ export const pageStyles: CSS.Properties = {
 
 export const NUMERICAL_FIELDS = ['short', 'integer', 'long', 'float', 'double'];
 
-export const ENABLED_VIS_TYPES = ['bar', 'horizontal_bar', 'line', 'pie', 'heatmap', 'text'];
+const { Bar, HorizontalBar, Line, Pie, HeatMap, Text} = visChartTypes;
+export const ENABLED_VIS_TYPES = [Bar, HorizontalBar, Line, Pie, HeatMap, Text];
```
[the remark this answers is not in the document]
Comment (rinku-kumar-psl) [reply dashboards-observability/common/constants/shared.ts]: Done, Thanks Subrat!

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx (hunk 7 above)
Comment (rinku-kumar-psl) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx]: Done, Thanks Amruta!

**Review by abasatwar — APPROVED** [review]
